File: packages/shared-types/src/dtos/boards/column.dto.ts
import { EntityTimestamps } from '../../types/common.types';

/**
 * Column DTO
 */
export interface ColumnDto extends EntityTimestamps {
  id: string;
  name: string;
  position: number;
  boardId: string;
  wipLimit: number | null;
}

/**
 * Create column request
 */
export interface ColumnCreateRequestDto {
  name: string;
  position?: number;
  boardId: string;
  wipLimit?: number;
}

/**
 * Update column request
 */
export interface ColumnUpdateRequestDto {
  name?: string;
  position?: number;
  wipLimit?: number;
}

File: apps/tui/internal/application/dto/column_dto.go
package dto

type ColumnDto struct {
	ID        string `json:"id"`
	Name      string `json:"name"`
	Position  int    `json:"position"`
	BoardID   string `json:"boardId"`
	WipLimit  *int   `json:"wipLimit"`
	CreatedAt string `json:"createdAt"`
	UpdatedAt string `json:"updatedAt"`
}

type ColumnCreateRequest struct {
	Name     string `json:"name"`
	BoardID  string `json:"boardId"`
	WipLimit *int   `json:"wipLimit,omitempty"`
}
